pkg/configuration/base/resources/proxy.go

```diff
@@ -0,0 +1,124 @@
+package resources
+
+import (
+	"context"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	openshiftGlobalProxy = "cluster"
+	envHTTPProxyName     = "HTTP_PROXY"
+	envHTTPSProxyName    = "HTTPS_PROXY"
+	envNoProxyName       = "NO_PROXY"
+)
+
+var (
+	ProxyAPIAvailable   = false
+	ProxyAPIChecked     = false
+	allProxyEnvVarNames = []string{
+		envHTTPProxyName,
+		envHTTPSProxyName,
+		envNoProxyName,
+	}
+	ProxyEnvVars = []corev1.EnvVar{}
+)
+
+func IsOpenshiftProxyAPIAvailable(clientSet *kubernetes.Clientset) bool {
+	if ProxyAPIChecked {
+		return ProxyAPIAvailable
+	}
+	gv := schema.GroupVersion{
+		Group:   apiconfigv1.GroupName,
+		Version: apiconfigv1.SchemeGroupVersion.Version,
+	}
+	if err := discovery.ServerSupportsVersion(clientSet, gv); err != nil {
+		// error, API not available
+		ProxyAPIChecked = true
+		ProxyAPIAvailable = false
```

Comment:
You can also set it to false before the if/else, and remove the else.

Reply:
done